include/boost/geometry/algorithms/detail/overlay/get_turn_info_for_endpoint.hpp

```diff
@@ -443,14 +413,14 @@ struct get_turn_info_for_endpoint
                 }
                 else if ( ip_j2 )
                 {
-                    side_calculator_for_endpoint<cs_tag,
-                                    RobustPoint1, RobustPoint2,
-                                    typename IntersectionInfo::side_strategy_type,
-                                    RobustPoint2>
-                        side_calc(ri2, ri1, rj1, ri2, rj2, rk2, inters.get_side_strategy()); // TODO: wrong, ri2/ri1/rj2
+                    // TODO: find out why side is used with respect to non-segments
+                    // (that is: range2.at(0) to range1.at(0), denoted as x)
+                    int const side_pj_q2 = sides.apply(range2.at(1), range2.at(2), range1.at(1));
+                    int const side_pj_x = sides.apply(range2.at(0), range1.at(0), range1.at(1));
+                    int const side_qk_x = sides.apply(range2.at(0), range1.at(0), range2.at(2));
```

Comment:
It is probably me who wrote it. I don't remember exactly why turns are checked this way, I'd have to analyse the whole file and maybe debug. However I may point you in a right direction right now. This is probably for turns where linear geometries are involved, correct? With linear geometries the turns have slightly different meaning than in areal/areal case. For instance the equality doesn't mean that the whole segment was the same but that the intersection point was equal to one of the points of the geometry. Turns hold additional data indicating whether prior to the intersection point the segments were collinear or are crossing at the intersection point, etc. AFAIR this had something to do with the fact that linear geometries are not ordered like areal and that from various geometrically equal sets of segments, I needed the same set of turns, no matter the direction of the segments. In other words the linear turns are non-ordered like areal, turns. E.g. for areal/areal one relative direction could result in i/i operations but with reverse segments the operations would be x/x. And this is true for all other operations. But linear turns would be the same.

So AFAIR parts of this code are there to transform ordered areal turns into unordered linear turns. Sometimes additional info had to be calculated. I didn't check this for sure but this might have been such case.

Reply:
Furthermore, this is the code calculating turns for endpoints, i.e. the first or the last point of a linestring. There is no such thing in areal geometries. However in this case we have to know how the other geometry behaves at the endpoint, when one geometry ends or begins. And since the geometry either begins or ends at intersection point maybe this is why "non-segment" points from other geometries are considered.